core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -621,15 +616,15 @@ class ReplicaManager(val config: KafkaConfig,
    *  To avoid ISR thrashing, we only throttle a replica on the leader if it's in the throttled replica list,
    *  the quota is exceeded and the replica is not in sync.
    */
-  def shouldLeaderThrottle(quota: ReplicaQuota, topicPartition: TopicAndPartition, replicaId: Int): Boolean = {
+  def shouldLeaderThrottle(quota: ReplicaQuota, topicPartition: TopicPartition, replicaId: Int): Boolean = {
     val isReplicaInSync = getPartition(topicPartition.topic, topicPartition.partition).flatMap { partition =>
       partition.getReplica(replicaId).map(partition.inSyncReplicas.contains)
     }.getOrElse(false)
     quota.isThrottled(topicPartition) && quota.isQuotaExceeded && !isReplicaInSync
   }
 
-  def getMagicAndTimestampType(topicAndPartition: TopicAndPartition): Option[(Byte, TimestampType)] =
-    getReplica(topicAndPartition.topic, topicAndPartition.partition).flatMap { replica =>
+  def getMagicAndTimestampType(topicPartition: TopicPartition): Option[(Byte, TimestampType)] =
+    getReplica(topicPartition.topic, topicPartition.partition).flatMap { replica =>
```

Comment:
It's a little weird that the `allPartitions` map in this class is keyed by a tuple instead of `TopicPartition`. Think it's worth changing here?